aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -344,10 +344,23 @@ Tensor real(const Tensor& self) {
   }
 }
 
+Tensor neg_view(const Tensor& self) {
+  Tensor self_;
+  auto impl = c10::make_intrusive<TensorImpl>(
+    Storage(self.storage()), self.key_set(), self.dtype());
+  impl->set_storage_offset(self.storage_offset());
+  impl->set_sizes_and_strides(self.sizes(), self.strides());
+  impl->set_neg(!self.is_neg());
+  self_ = Tensor(std::move(impl));
+  namedinference::propagate_names(self_, self);
+  return self_;
+}
+
 Tensor imag(const Tensor& self) {
   if (self.is_complex()) {
-    auto real_tensor = at::view_as_real(self);
-    return at::select(real_tensor, real_tensor.dim() - 1, 1);
+    auto real_tensor = at::_view_as_real_physical(self);
+    auto true_real_tensor = self.is_conj() ? at::neg_view(real_tensor) : real_tensor;
+    return at::select(true_real_tensor, real_tensor.dim() - 1, 1);
```

Comment:
nit: this is a preexisting condition, but it ought to be possible to skip redundant redispatches here

Reply:
there doesn't seem to be a dispatch for `select` (forward).